### ISSUE TITLE
Is POST of entityOperations/upsert deleting the entity temporarily ?

### ISSUE THREAD

**Issue by pdmangel2** [Fiware/context.Orion-LD#1296]
Hi,

in order to update a full entity, i.e. not only adding, updating existing attributes, but also deleting attributes that are not present in the new version, we implemented a POST on entityOperations/upsert endpoint.
I already faced the problem in the past that sometimes a GET on entity id, or a query didn't return an existing entity.
Today I noticed that when this happens I also see some hundred milliseconds before an upsert on that same entity.
Since these actions are not synchronized, they might be overlapping.
Is it possible that the entityOperations/upsert deletes the entity and recreates it ?

**Comment by kzangeli** [comment]
Yes, I believe I implemented it that way. 
But, the "createdAt" timestamp is preserved (taken from the deleted entity). And "type" is not allowed to be modified

**Comment by pdmangel2** [comment]
Thanks for the clarification, Ken.
This might explain why we sometimes see existing entities temporarily disappearing in queries that coincide with such an upsert.
Is there any way to fix this behaviour (e.g. using a mongo upsert) ? We now implemeted retries on the 'get by id', but can't do that on a query.
Or is there an alternative to update the full entity in one operation ?

**Comment by kzangeli** [comment]
Yeah, I see that this can be a problem. Sorry, didn't think about that when I decided to implement it that way.
I will look into it
